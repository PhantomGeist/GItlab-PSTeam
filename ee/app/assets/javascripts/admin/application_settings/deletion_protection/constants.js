import { __, s__ } from '~/locale';

export const I18N_DELETION_PROTECTION = {
  label: s__('DeletionSettings|Deletion protection'),
  helpText: s__(
    'DeletionSettings|Period that deleted groups and projects will remain restorable for. Personal projects are always deleted immediately.',
  ),
  learnMore: __('Learn more.'),
  days: __('days'),
};
