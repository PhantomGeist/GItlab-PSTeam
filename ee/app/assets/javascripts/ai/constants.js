import { s__, sprintf } from '~/locale';

export const i18n = {
  GITLAB_DUO: s__('AI|GitLab Duo'),
  GENIE_TOOLTIP: s__('AI|What does the selected code mean?'),
  GENIE_NO_CONTAINER_ERROR: s__("AI|The container element wasn't found, stopping AI Genie."),
  GENIE_CHAT_TITLE: s__('AI|Code Explanation'),
  GENIE_CHAT_CLOSE_LABEL: s__('AI|Close the Code Explanation'),
  GENIE_CHAT_LEGAL_NOTICE: sprintf(
    s__(
      'AI|You are not allowed to copy any part of this output into issues, comments, GitLab source code, commit messages, merge requests or any other user interface in the %{gitlabOrg} or %{gitlabCom} groups.',
    ),
    { gitlabOrg: '<code>/gitlab-org</code>', gitlabCom: '<code>/gitlab-com</code>' },
    false,
  ),
  GENIE_CHAT_LEGAL_GENERATED_BY_AI: s__('AI|Responses generated by AI'),
  REQUEST_ERROR: s__('AI|Something went wrong. Please try again later'),
  EXPERIMENT_BADGE: s__('AI|Experiment'),
  EXPERIMENT_POPOVER_TITLE: s__("AI|What's an Experiment?"),
  EXPERIMENT_POPOVER_CONTENT: s__(
    "AI|An %{linkStart}Experiment%{linkEnd} is a feature that's in the process of being developed. It's not production-ready. We encourage users to try Experimental features and provide feedback. An Experiment: %{bullets}",
  ),
  EXPERIMENT_POPOVER_BULLETS: [
    s__('AI|May be unstable'),
    s__('AI|Has no support and might not be documented'),
    s__('AI|Can be removed at any time'),
  ],
  EXPLAIN_CODE_PROMPT: s__(
    'AI|Explain the code from %{filePath} in human understandable language presented in Markdown format. In the response add neither original code snippet nor any title. `%{text}`. If it is not programming code, say `The selected text is not code. I am afraid this feature is for explaining code only. Would you like to ask a different question about the selected text?` and wait for another question.',
  ),
  TOO_LONG_ERROR_MESSAGE: s__(
    'AI|There is too much text in the chat. Please try again with a shorter text.',
  ),
  GENIE_CHAT_PROMPT_PLACEHOLDER: s__('AI|GitLab Duo Chat'),
  GENIE_CHAT_SUBMIT_LABEL: s__('AI|Send chat message.'),
  GENIE_CHAT_EMPTY_STATE_TITLE: s__('AI|Ask a question'),
  GENIE_CHAT_EMPTY_STATE_DESC: s__('AI|AI generated explanations will appear here.'),
  GENIE_CHAT_LEGAL_DISCLAIMER: s__(
    "AI|May provide inappropriate responses not representative of GitLab's views. Do not input personal data.",
  ),
  GENIE_CHAT_NEW_CHAT: s__('AI|New chat'),
  GENIE_CHAT_LOADING_MESSAGE: s__('AI|%{tool} is %{transition} an answer'),
  GENIE_CHAT_LOADING_TRANSITIONS: [
    s__('AI|finding'),
    s__('AI|working on'),
    s__('AI|generating'),
    s__('AI|producing'),
  ],
  GENIE_CHAT_FEEDBACK_LINK: s__('AI|Give feedback to improve this answer.'),
  GENIE_CHAT_FEEDBACK_THANKS: s__('AI|Thank you for your feedback.'),
};
export const GENIE_CHAT_LOADING_TRANSITION_DURATION = 7500;
export const TOO_LONG_ERROR_TYPE = 'too-long';
export const AI_GENIE_DEBOUNCE = 300;
export const GENIE_CHAT_MODEL_ROLES = {
  user: 'user',
  system: 'system',
  assistant: 'assistant',
};

export const CHAT_MESSAGE_TYPES = {
  tool: 'tool',
};

export const FEEDBACK_OPTIONS = [
  {
    title: s__('AI|Helpful'),
    icon: 'thumb-up',
    value: 'helpful',
  },
  {
    title: s__('AI|Unhelpful'),
    icon: 'thumb-down',
    value: 'unhelpful',
  },
  {
    title: s__('AI|Wrong'),
    icon: 'status_warning',
    value: 'wrong',
  },
];

export const EXPLAIN_CODE_TRACKING_EVENT_NAME = 'explain_code_blob_viewer';
export const TANUKI_BOT_TRACKING_EVENT_NAME = 'ask_gitlab_chat';
export const GENIE_CHAT_RESET_MESSAGE = '/reset';

export const DOCUMENTATION_SOURCE_TYPES = {
  HANDBOOK: {
    value: 'handbook',
    icon: 'book',
  },
  DOC: {
    value: 'doc',
    icon: 'documents',
  },
  BLOG: {
    value: 'blog',
    icon: 'list-bulleted',
  },
};
