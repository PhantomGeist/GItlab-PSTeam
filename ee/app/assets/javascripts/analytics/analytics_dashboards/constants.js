import { s__ } from '~/locale';

export const FEATURE_PRODUCT_ANALYTICS = 'productAnalytics';

export const PRODUCT_ANALYTICS_FEATURE_DASHBOARDS = ['audience', 'behavior'];

export const EVENTS_TYPES = ['pageViews', 'linkClickEvents', 'events'];

export function isTrackedEvent(eventType) {
  return EVENTS_TYPES.includes(eventType);
}

export const PANEL_VISUALIZATION_HEIGHT = '600px';

export const PANEL_DISPLAY_TYPES = {
  DATA: 'data',
  VISUALIZATION: 'visualization',
  CODE: 'code',
};

export const PANEL_DISPLAY_TYPE_ITEMS = [
  {
    type: PANEL_DISPLAY_TYPES.DATA,
    icon: 'table',
    title: s__('Analytics|Data'),
  },
  {
    type: PANEL_DISPLAY_TYPES.VISUALIZATION,
    icon: 'chart',
    title: s__('Analytics|Visualization'),
  },
  {
    type: PANEL_DISPLAY_TYPES.CODE,
    icon: 'code',
    title: s__('Analytics|Code'),
  },
];

export const MEASURE_COLOR = '#00b140';
export const DIMENSION_COLOR = '#c3e6cd';

export const EVENTS_TABLE_NAME = 'TrackedEvents';
export const SESSIONS_TABLE_NAME = 'Sessions';

export const ANALYTICS_FIELD_CATEGORIES = [
  {
    name: s__('Analytics|Pages'),
    category: 'pages',
  },
  {
    name: s__('Analytics|Users'),
    category: 'users',
  },
  {
    name: s__('Analytics|Link clicks'),
    category: 'linkClicks',
  },
  {
    name: s__('Analytics|Custom events'),
    category: 'custom',
  },
];

export const ANALYTICS_FIELDS = [
  {
    name: s__('Analytics|URL'),
    category: 'pages',
    dbField: 'pageUrl',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Page Path'),
    category: 'pages',
    dbField: 'pageUrlpath',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Page Title'),
    category: 'pages',
    dbField: 'pageTitle',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Page Language'),
    category: 'pages',
    dbField: 'documentLanguage',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Host'),
    category: 'pages',
    dbField: 'pageUrlhosts',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Referer'),
    category: 'users',
    dbField: 'pageReferrer',
    icon: 'user',
  },
  {
    name: s__('Analytics|Language'),
    category: 'users',
    dbField: 'browserLanguage',
    icon: 'user',
  },
  {
    name: s__('Analytics|Viewport'),
    category: 'users',
    dbField: 'viewportSize',
    icon: 'user',
  },
  {
    name: s__('Analytics|Browser Family'),
    category: 'users',
    dbField: 'agentName',
    icon: 'user',
  },
  {
    name: s__('Analytics|Browser'),
    category: 'users',
    dbField: ['agentName', 'agentVersion'],
    icon: 'user',
  },
  {
    name: s__('Analytics|OS'),
    category: 'users',
    dbField: 'osName',
    icon: 'user',
  },
  {
    name: s__('Analytics|OS Version'),
    category: 'users',
    dbField: ['osName', 'osVersion'],
    icon: 'user',
  },
  {
    name: s__('Analytics|User Id'),
    category: 'users',
    dbField: 'userId',
    icon: 'user',
  },
  {
    name: s__('Analytics|Target URL'),
    category: 'linkClicks',
    dbField: ['targetUrl'],
    icon: 'link',
  },
  {
    name: s__('Analytics|Element ID'),
    category: 'linkClicks',
    dbField: ['elementId'],
    icon: 'link',
  },
  {
    name: s__('Analytics|Event Name'),
    category: 'custom',
    dbField: 'customEventName',
    icon: 'documents',
  },
  {
    name: s__('Analytics|Event Props'),
    category: 'custom',
    dbField: 'customEventProps',
    icon: 'documents',
  },
  {
    name: s__('Analytics|User Props'),
    category: 'custom',
    dbField: 'customUserProps',
    icon: 'user',
  },
];

export const NEW_DASHBOARD = () => ({
  title: '',
  description: '',
  panels: [],
  userDefined: true,
});

export const FILE_ALREADY_EXISTS_SERVER_RESPONSE = 'A file with this name already exists';
