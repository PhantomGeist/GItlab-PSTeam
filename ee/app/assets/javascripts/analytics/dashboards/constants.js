import { green400, orange400, red400, gray400 } from '@gitlab/ui/scss_to_js/scss_variables';
import { s__, __, n__, sprintf } from '~/locale';
import { helpPagePath } from '~/helpers/help_page_helper';
import {
  FLOW_METRICS,
  DORA_METRICS,
  VULNERABILITY_METRICS,
  MERGE_REQUEST_METRICS,
} from '~/analytics/shared/constants';

export const UNITS = {
  COUNT: 'COUNT',
  DAYS: 'DAYS',
  PER_DAY: 'PER_DAY',
  PERCENT: 'PERCENT',
};

export const SUPPORTED_DORA_METRICS = [
  DORA_METRICS.DEPLOYMENT_FREQUENCY,
  DORA_METRICS.LEAD_TIME_FOR_CHANGES,
  DORA_METRICS.TIME_TO_RESTORE_SERVICE,
  DORA_METRICS.CHANGE_FAILURE_RATE,
];

export const SUPPORTED_FLOW_METRICS = [
  FLOW_METRICS.LEAD_TIME,
  FLOW_METRICS.CYCLE_TIME,
  FLOW_METRICS.ISSUES,
  FLOW_METRICS.ISSUES_COMPLETED,
  FLOW_METRICS.DEPLOYS,
];

export const SUPPORTED_MERGE_REQUEST_METRICS = [MERGE_REQUEST_METRICS.THROUGHPUT];

export const SUPPORTED_VULNERABILITY_METRICS = [
  VULNERABILITY_METRICS.CRITICAL,
  VULNERABILITY_METRICS.HIGH,
];

export const TABLE_METRICS = {
  [DORA_METRICS.DEPLOYMENT_FREQUENCY]: {
    label: s__('DORA4Metrics|Deployment Frequency'),
    units: UNITS.PER_DAY,
  },
  [DORA_METRICS.LEAD_TIME_FOR_CHANGES]: {
    label: s__('DORA4Metrics|Lead Time for Changes'),
    invertTrendColor: true,
    units: UNITS.DAYS,
  },
  [DORA_METRICS.TIME_TO_RESTORE_SERVICE]: {
    label: s__('DORA4Metrics|Time to Restore Service'),
    invertTrendColor: true,
    units: UNITS.DAYS,
  },
  [DORA_METRICS.CHANGE_FAILURE_RATE]: {
    label: s__('DORA4Metrics|Change Failure Rate'),
    invertTrendColor: true,
    units: UNITS.PERCENT,
  },
  [FLOW_METRICS.LEAD_TIME]: {
    label: s__('DORA4Metrics|Lead time'),
    invertTrendColor: true,
    units: UNITS.DAYS,
  },
  [FLOW_METRICS.CYCLE_TIME]: {
    label: s__('DORA4Metrics|Cycle time'),
    invertTrendColor: true,
    units: UNITS.DAYS,
  },
  [FLOW_METRICS.ISSUES]: {
    label: s__('DORA4Metrics|Issues created'),
    units: UNITS.COUNT,
  },
  [FLOW_METRICS.ISSUES_COMPLETED]: {
    label: s__('DORA4Metrics|Issues closed'),
    units: UNITS.COUNT,
    valueLimit: {
      max: 10001,
      mask: '10000+',
      description: s__(
        'DORA4Metrics|This is a lower-bound approximation. Your group has too many issues and MRs to calculate in real time.',
      ),
    },
  },
  [FLOW_METRICS.DEPLOYS]: {
    label: s__('DORA4Metrics|Deploys'),
    units: UNITS.COUNT,
  },
  [MERGE_REQUEST_METRICS.THROUGHPUT]: {
    label: s__('DORA4Metrics|Merge request throughput'),
    units: UNITS.COUNT,
  },
  [VULNERABILITY_METRICS.CRITICAL]: {
    label: s__('DORA4Metrics|Critical Vulnerabilities over time'),
    invertTrendColor: true,
    units: UNITS.COUNT,
  },
  [VULNERABILITY_METRICS.HIGH]: {
    label: s__('DORA4Metrics|High Vulnerabilities over time'),
    invertTrendColor: true,
    units: UNITS.COUNT,
  },
};

export const METRICS_WITH_NO_TREND = [VULNERABILITY_METRICS.CRITICAL, VULNERABILITY_METRICS.HIGH];
export const METRICS_WITH_LABEL_FILTERING = [
  FLOW_METRICS.ISSUES,
  FLOW_METRICS.ISSUES_COMPLETED,
  FLOW_METRICS.CYCLE_TIME,
  FLOW_METRICS.LEAD_TIME,
  MERGE_REQUEST_METRICS.THROUGHPUT,
];
export const METRICS_WITHOUT_LABEL_FILTERING = Object.keys(TABLE_METRICS).filter(
  (metric) => !METRICS_WITH_LABEL_FILTERING.includes(metric),
);

export const DASHBOARD_TITLE = s__('DORA4Metrics|Value Streams Dashboard');
export const DASHBOARD_DESCRIPTION = s__(
  'DORA4Metrics|The Value Streams Dashboard allows all stakeholders from executives to individual contributors to identify trends, patterns, and opportunities for software development improvements.',
);
export const DASHBOARD_SURVEY_LINK =
  'https://gitlab.fra1.qualtrics.com/jfe/form/SV_50guMGNU2HhLeT4';
export const DASHBOARD_DOCS_LINK = helpPagePath('user/analytics/value_streams_dashboard');
export const DASHBOARD_DESCRIPTION_GROUP = s__('DORA4Metrics|Metrics comparison for %{name} group');
export const DASHBOARD_DESCRIPTION_PROJECT = s__(
  'DORA4Metrics|Metrics comparison for %{name} project',
);
export const DASHBOARD_LOADING_FAILURE = s__('DORA4Metrics|Some metric comparisons failed to load');
export const DASHBOARD_NAMESPACE_LOAD_ERROR = s__(
  'DORA4Metrics|Failed to load comparison chart for Namespace: %{fullPath}',
);
export const DASHBOARD_LABELS_LOAD_ERROR = s__(
  'DORA4Metrics|Failed to load labels matching the filter: %{labels}',
);
export const VALUE_STREAMS_DASHBOARD_CONFIG = {
  title: DASHBOARD_TITLE,
  description: DASHBOARD_DESCRIPTION,
  slug: '/value_streams_dashboard',
  redirect: true,
  userDefined: false,
};

export const CHART_GRADIENT = ['#499767', '#5252B5'];
export const CHART_GRADIENT_INVERTED = [...CHART_GRADIENT].reverse();
export const CHART_LOADING_FAILURE = s__('DORA4Metrics|Some metric charts failed to load');

export const CHART_TOOLTIP_UNITS = {
  [UNITS.COUNT]: undefined,
  [UNITS.DAYS]: __('days'),
  [UNITS.PER_DAY]: __('/day'),
  [UNITS.PERCENT]: '%',
};

export const YAML_CONFIG_PATH = '.gitlab/analytics/dashboards/value_streams/value_streams.yaml';
export const YAML_CONFIG_LOAD_ERROR = s__(
  'DORA4Metrics|Failed to load YAML config from Project: %{fullPath}',
);

export const ALERT_TEXT = s__(
  'DORA4Metrics|To help us improve the Value Stream Management Dashboard, please share feedback about your experience in this %{linkStart}survey%{linkEnd}.',
);

export const CLICK_METRIC_DRILLDOWN_LINK_ACTION = 'click_link';

export const DORA_PERFORMERS_SCORE_CATEGORY_TYPES = {
  HIGH: 'highProjectsCount',
  MEDIUM: 'mediumProjectsCount',
  LOW: 'lowProjectsCount',
  NO_DATA: 'noDataProjectsCount',
};

export const DORA_PERFORMERS_SCORE_CATEGORIES = {
  [DORA_PERFORMERS_SCORE_CATEGORY_TYPES.HIGH]: s__('DORA4Metrics|High'),
  [DORA_PERFORMERS_SCORE_CATEGORY_TYPES.MEDIUM]: s__('DORA4Metrics|Medium'),
  [DORA_PERFORMERS_SCORE_CATEGORY_TYPES.LOW]: s__('DORA4Metrics|Low'),
  [DORA_PERFORMERS_SCORE_CATEGORY_TYPES.NO_DATA]: s__('DORA4Metrics|Not included'),
};

export const DORA_PERFORMERS_SCORE_METRICS = [
  // score definitions are listed in order from 'High' to 'Low' and accessed using the series index
  {
    label: s__('DORA4Metrics|Deployment Frequency (Velocity)'),
    scoreDefinitions: [
      s__('DORA4Metrics|Have 30 or more deploys to production per day.'),
      s__('DORA4Metrics|Have between 1 to 29 deploys to production per day.'),
      s__('DORA4Metrics|Have less than 1 deploy to production per day.'),
    ],
  },
  {
    label: s__('DORA4Metrics|Lead Time for Changes (Velocity)'),
    scoreDefinitions: [
      s__(
        'DORA4Metrics|Took 7 days or less to go from code committed to code successfully running in production.',
      ),
      s__(
        'DORA4Metrics|Took between 8 to 29 days to go from code committed to code successfully running in production.',
      ),
      s__(
        'DORA4Metrics|Took more than 30 days to go from code committed to code successfully running in production.',
      ),
    ],
  },
  {
    label: s__('DORA4Metrics|Time to Restore Service (Quality)'),
    scoreDefinitions: [
      s__(
        'DORA4Metrics|Took 1 day or less to restore service when a service incident or a defect that impacts users occurs.',
      ),
      s__(
        'DORA4Metrics|Took between 2 to 6 days to restore service when a service incident or a defect that impacts users occurs.',
      ),
      s__(
        'DORA4Metrics|Took more than 7 days to restore service when a service incident or a defect that impacts users occurs.',
      ),
    ],
  },
  {
    label: s__('DORA4Metrics|Change Failure Rate (Quality)'),
    scoreDefinitions: [
      sprintf(
        s__('DORA4Metrics|Made 15%% or less changes to production resulted in degraded service.'),
      ),
      sprintf(
        s__(
          'DORA4Metrics|Made between 16%% to 44%% of changes to production resulted in degraded service.',
        ),
      ),
      sprintf(
        s__(
          'DORA4Metrics|Made more than 45%% of changes to production resulted in degraded service.',
        ),
      ),
    ],
  },
];

export const DORA_PERFORMERS_SCORE_DEFAULT_PANEL_TITLE = s__(
  'DORA4Metrics|Total projects with DORA performers score',
);

export const DORA_PERFORMERS_SCORE_PANEL_TITLE_WITH_PROJECTS_COUNT = s__(
  'DORA4Metrics|Total projects (%{count}) with DORA performers score for %{groupName} group',
);

export const DORA_PERFORMERS_SCORE_TOOLTIP_PROJECTS_COUNT_TITLE = (count) =>
  n__('DORA4Metrics|%d project', 'DORA4Metrics|%d projects', count);

export const DORA_PERFORMERS_SCORE_NOT_INCLUDED = (count) =>
  n__('DORA4Metrics|Has no calculated data.', 'DORA4Metrics|Have no calculated data.', count);

export const DORA_PERFORMERS_SCORE_LOADING_ERROR = s__(
  'DORA4Metrics|Failed to load DORA performance scores for Namespace: %{fullPath}',
);

export const DORA_PERFORMERS_SCORE_PROJECT_NAMESPACE_ERROR = s__(
  'DORA4Metrics|This visualization is not supported for project namespaces.',
);

export const DORA_PERFORMERS_SCORE_NO_DATA = s__(
  'DORA4Metrics|No data available for Namespace: %{fullPath}',
);

export const DORA_PERFORMERS_SCORE_CHART_COLOR_PALETTE = [green400, orange400, red400, gray400];
