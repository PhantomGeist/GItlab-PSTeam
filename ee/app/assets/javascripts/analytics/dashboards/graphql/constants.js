export const BUCKETING_INTERVAL_ALL = 'ALL';

export const MERGE_REQUESTS_STATE_MERGED = 'merged';
