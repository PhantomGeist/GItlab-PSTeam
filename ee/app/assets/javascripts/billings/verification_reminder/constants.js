import { s__ } from '~/locale';
import { DOCS_URL } from 'jh_else_ce/lib/utils/url_utility';

export const FEATURE_NAME = 'verification_reminder';
export const DOCS_LINK = `${DOCS_URL}/runner/install/`;
export const EVENT_LABEL = 'verification_reminder';
export const MOUNTED_EVENT = 'shown';
export const DISMISS_EVENT = 'dismissed';
export const OPEN_DOCS_EVENT = 'clicked_docs_link';
export const START_VERIFICATION_EVENT = 'start_verification';
export const SUCCESSFUL_VERIFICATION_EVENT = 'successful_verification';
export const I18N = {
  warningAlert: {
    title: s__(
      'VerificationReminder|Pipeline failing? To keep GitLab spam and abuse free we ask that you verify your identity.',
    ),
    message: s__(
      'VerificationReminder|Until then, shared runners will be unavailable. %{validateLinkStart}Validate your account%{validateLinkEnd} or %{docsLinkStart}use your own runners%{docsLinkEnd}.',
    ),
  },
  successAlert: {
    title: s__('VerificationReminder|Your account has been validated'),
    message: s__(
      'VerificationReminder|Youâ€™ll now be able to take advantage of free compute minutes on shared runners.',
    ),
  },
};
