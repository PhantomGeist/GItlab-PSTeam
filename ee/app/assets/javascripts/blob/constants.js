import { n__, s__ } from '~/locale';
import { DOCS_URL_IN_EE_DIR } from 'jh_else_ce/lib/utils/url_utility';

export const SYNTAX_ERRORS_TEXT = (count) =>
  n__(
    'CodeownersValidation|Contains %d syntax error.',
    'CodeownersValidation|Contains %d syntax errors.',
    count,
  );

export const SYNTAX_VALID = s__('CodeownersValidation|Syntax is valid.');

export const DOCS_LINK_TEXT = s__('CodeownersValidation|How are errors handled?');

export const SHOW_ERRORS = s__('CodeownersValidation|Show errors');

export const HIDE_ERRORS = s__('CodeownersValidation|Hide errors');

export const COLLAPSE_ID = 'CODEOWNERS_VALIDATION_COLLAPSE';

export const LINE = s__('CodeownersValidation|Line');

export const ERROR_MESSAGE = s__(
  'CodeownersValidation|An error occurred while loading the validation errors. Please try again later.',
);

export const DOCS_URL = `${DOCS_URL_IN_EE_DIR}/user/project/codeowners/reference.html#error-handling-in-code-owners`;

export const CODEOWNERS_VALIDATION_I18N = {
  syntaxValid: SYNTAX_VALID,
  syntaxErrors: SYNTAX_ERRORS_TEXT,
  show: SHOW_ERRORS,
  hide: HIDE_ERRORS,
  docsLink: DOCS_LINK_TEXT,
  line: LINE,
  errorMessage: ERROR_MESSAGE,
};

export const INVALID_SECTION_OWNER_FORMAT = s__('CodeownersValidation|Inaccessible owners');

export const MISSING_ENTRY_OWNER = s__('CodeownersValidation|Zero owners');

export const INVALID_ENTRY_OWNER_FORMAT = s__('CodeownersValidation|Entries with spaces');

export const MISSING_SECTION_NAME = s__('CodeownersValidation|Missing section name');

export const INVALID_APPROVAL_REQUIREMENT = s__(
  'CodeownersValidation|Less than 1 required approvals',
);

export const INVALID_SECTION_FORMAT = s__('CodeownersValidation|Unparsable sections');

export const CODE_TO_MESSAGE = {
  invalid_section_owner_format: INVALID_SECTION_OWNER_FORMAT,
  missing_entry_owner: MISSING_ENTRY_OWNER,
  invalid_entry_owner_format: INVALID_ENTRY_OWNER_FORMAT,
  missing_section_name: MISSING_SECTION_NAME,
  invalid_approval_requirement: INVALID_APPROVAL_REQUIREMENT,
  invalid_section_format: INVALID_SECTION_FORMAT,
};
