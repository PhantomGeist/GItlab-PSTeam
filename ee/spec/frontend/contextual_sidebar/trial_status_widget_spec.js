import { GlLink } from '@gitlab/ui';
import { shallowMountExtended } from 'helpers/vue_test_utils_helper';
import { WIDGET } from 'ee/contextual_sidebar/components/constants';
import TrialStatusWidget from 'ee/contextual_sidebar/components/trial_status_widget.vue';
import { mockTracking, unmockTracking } from 'helpers/tracking_helper';

describe('TrialStatusWidget component', () => {
  let wrapper;

  const { trackingEvents } = WIDGET;
  const trialDaysUsed = 10;
  const trialDuration = 30;

  const findGlLink = () => wrapper.findComponent(GlLink);

  const createComponent = (providers = {}) => {
    return shallowMountExtended(TrialStatusWidget, {
      provide: {
        trialDaysUsed,
        trialDuration,
        navIconImagePath: 'illustrations/golden_tanuki.svg',
        percentageComplete: 10,
        planName: 'Ultimate',
        plansHref: 'billing/path-for/group',
        ...providers,
      },
    });
  };

  describe('interpolated strings', () => {
    it('correctly interpolates them all', () => {
      wrapper = createComponent();

      expect(wrapper.text()).not.toMatch(/%{\w+}/);
    });
  });

  describe('without the optional containerId prop', () => {
    beforeEach(() => {
      wrapper = createComponent();
    });

    it('matches the snapshot for namespace in active trial', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('matches the snapshot for namespace not in active trial', () => {
      wrapper = createComponent({ percentageComplete: 110 });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders without an id', () => {
      expect(findGlLink().attributes('id')).toBe(undefined);
    });

    describe('tracks when the widget menu is clicked', () => {
      let trackingSpy;

      beforeEach(() => {
        trackingSpy = mockTracking(undefined, undefined, jest.spyOn);
      });

      afterEach(() => {
        unmockTracking();
      });

      it('tracks with correct information when namespace is in an active trial', async () => {
        const { category, label } = trackingEvents.activeTrialOptions;
        await wrapper.findByTestId('widget-menu').trigger('click');

        expect(trackingSpy).toHaveBeenCalledWith(category, trackingEvents.action, {
          category,
          label,
        });
      });

      it('tracks with correct information when namespace is not in an active trial', async () => {
        wrapper = createComponent({ percentageComplete: 110 });

        const { category, label } = trackingEvents.trialEndedOptions;
        await wrapper.findByTestId('widget-menu').trigger('click');

        expect(trackingSpy).toHaveBeenCalledWith(category, trackingEvents.action, {
          category,
          label,
        });
      });
    });

    it('does not render Trial twice if the plan name includes "Trial"', () => {
      wrapper = createComponent({ planName: 'Ultimate Trial' });

      expect(wrapper.text()).toMatchInterpolatedText('Ultimate Trial Day 10/30');
    });

    it('shows the expected day 1 text', () => {
      wrapper = createComponent({ trialDaysUsed: 1 });

      expect(wrapper.text()).toMatchInterpolatedText('Ultimate Trial Day 1/30');
    });

    it('shows the expected last day text', () => {
      wrapper = createComponent({ trialDaysUsed: 30 });

      expect(wrapper.text()).toMatchInterpolatedText('Ultimate Trial Day 30/30');
    });
  });

  describe('with the optional containerId prop', () => {
    beforeEach(() => {
      wrapper = createComponent({ containerId: 'some-id' });
    });

    it('renders with the given id', () => {
      expect(findGlLink().attributes('id')).toBe('some-id');
    });
  });
});
