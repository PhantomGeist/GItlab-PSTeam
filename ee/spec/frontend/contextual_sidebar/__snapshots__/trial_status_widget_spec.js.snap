// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrialStatusWidget component without the optional containerId prop matches the snapshot for namespace in active trial 1`] = `
<gl-link-stub
  href="billing/path-for/group"
  title="Ultimate Trial"
>
  <div
    class="gl-align-items-stretch gl-display-flex gl-flex-direction-column gl-w-full"
    data-testid="widget-menu"
  >
    <div>
      <div
        class="gl-display-flex gl-w-full"
      >
        <span
          class="gl-mr-3 nav-icon-container svg-container"
        >
          <img
            class="svg"
            src="illustrations/golden_tanuki.svg"
            width="16"
          />
        </span>
        <span
          class="gl-flex-grow-1 nav-item-name"
        >
          Ultimate Trial
        </span>
        <span
          class="collapse-text gl-font-sm gl-mr-auto"
        >
          Day 10/30
        </span>
      </div>
      <div
        class="gl-align-items-stretch gl-display-flex gl-mt-2"
      >
        <gl-progress-bar-stub
          aria-hidden="true"
          class="gl-flex-grow-1"
          value="10"
        />
      </div>
    </div>
  </div>
</gl-link-stub>
`;

exports[`TrialStatusWidget component without the optional containerId prop matches the snapshot for namespace not in active trial 1`] = `
<gl-link-stub
  href="billing/path-for/group"
  title="Your 30-day trial has ended"
>
  <div
    class="gl-align-items-stretch gl-display-flex gl-flex-direction-column gl-w-full"
    data-testid="widget-menu"
  >
    <div
      class="gl-display-flex gl-gap-5 gl-px-2 gl-w-full"
    >
      <gl-icon-stub
        class="gl-text-blue-600!"
        name="information-o"
        size="16"
      />
      <div>
        <div
          class="gl-font-weight-bold"
        >
          Your 30-day trial has ended
        </div>
        <div
          class="gl-mt-3"
        >
          Looking to do more with GitLab?
        </div>
      </div>
    </div>
  </div>
</gl-link-stub>
`;
